cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectBuild.java

```diff
@@ -324,35 +355,68 @@ private List<EnvVar> getBuildContainerEnvVars() {
     }
 
     /**
-     * Generates the builder container with the Kaniko executor
+     * Generates the builder container for Buildah or Kaniko.
      *
-     * @param imagePullPolicy   Image pull policy
+     * @param imagePullPolicy   Image pull policy.
+     * @param isBuildahBuild    Flag defining whether we should use Buildah or not.
+     *
+     * @return  Builder container definition which will be used in the Pod.
+     */
+    Container createContainer(ImagePullPolicy imagePullPolicy, boolean isBuildahBuild) {
+        return ContainerUtils.createContainer(
+            componentName,
+            image,
+            isBuildahBuild ? buildahArguments() : kanikoArguments(),
+            securityProvider.kafkaConnectBuildContainerSecurityContext(new ContainerSecurityProviderContextImpl(templateContainer)),
+            resources,
+            getBuildContainerEnvVars(),
+            null,
+            getVolumeMounts(),
+            null,
+            null,
+            imagePullPolicy
+        );
+    }
+
+    /**
+     * Returns Kaniko arguments needed for the build (and push) of the image - used inside the container.
      *
-     * @return  Builder container definition which will be used in the Pod
+     * @return Kaniko arguments needed for the build (and push) of the image - used inside the container.
      */
-    /* test */ Container createContainer(ImagePullPolicy imagePullPolicy) {
-        List<String> args = additionalKanikoOptions != null ? new ArrayList<>(4 + additionalKanikoOptions.size()) : new ArrayList<>(4);
+    List<String> kanikoArguments() {
+        List<String> args = additionalBuildOptions != null ? new ArrayList<>(4 + additionalBuildOptions.size()) : new ArrayList<>(4);
```

Comment:
shouldn't it be 3 enough given the 3 fixed arguments you add next? or why 4?

Reply:
I think yes, but TBH I just copied and pasted stuff from previous implementation :) but I can change it and try it with 3.